components/backup/src/writer.rs

```diff
@@ -65,17 +75,42 @@ impl BackupWriter {
                         // it, we need to add the prefix manually.
                         let data_key_default = keys::data_key(&default.0);
                         self.default.put(&data_key_default, &default.1)?;
-                        self.default_written = true;
+                        value_in_default.replace(true);
```

Comment:
```suggestion
                        value_in_default = Some(true);
```

No point using `.replace()` if you don't care about its return value.

Reply:
I think use `.replace` is more declarative.